src/file_utils.c

```diff
@@ -36,3 +36,38 @@ int create_file_name(char *new_name, const size_t name_size, const char *templat
 
 	return 0;
 }
+
+/**
+ * Get file size
+ * @param [in] f file handle
+ * @param [in] filename File name used to display the error message
+ * @param [out] size output for file size
+ * @param error code, 0 when success
+ */
+int get_file_size(FILE *f, const char* filename, size_t *size)
+{
+	int ret;
+
+	assert(size);
+
+	/* get file size */
+	ret = fseek(f, 0, SEEK_END);
+	if (ret < 0) {
+		fprintf(stderr, "error: unable to seek eof %s %d\n", filename, errno);
```

Comment:
fwiw, its really nice to use ```strerror(errno)``` in places like this. i.e. instead of printing the %d errno you can print the %s strerror(errno)